internal/cephfs/controllerserver.go

```diff
@@ -866,18 +852,8 @@ func (cs *ControllerServer) CreateSnapshot(
 
 	metadata := k8s.GetSnapshotMetadata(req.GetParameters())
 	if sid != nil {
-		// check snapshot is protected
-		protected := true
 		snapClient := core.NewSnapshot(parentVolOptions.GetConnection(), sid.FsSnapshotName,
```

Comment:
Move this client creation to inside if statement?

Reply:
that won't work, snapclient is also used online 885 (a little below this if)

Reply:
i dont think its used elsewhere inside the `if sid` the client is only used to set metadata for already found snapshot. please let me know if am missing something

Reply:
client creation can happen inside `if len(metadata) != 0 {`

Reply:
> i dont think its used elsewhere inside the `if sid` the client is only used to set metadata for already found snapshot. please let me know if am missing something

yes, thats correct. @nixpanic mentioned the same

> client creation can happen inside if len(metadata) != 0 {

I'll move under this if